packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts

```diff
@@ -154,15 +162,42 @@ export const listShippingOptionsForCartWorkflow = createWorkflow(
     }).config({ name: "shipping-options-query" })
 
     const shippingOptionsWithPrice = transform(
-      { shippingOptions },
-      ({ shippingOptions }) =>
+      { shippingOptions, cart },
+      ({ shippingOptions, cart }) =>
         shippingOptions.map((shippingOption) => {
           const price = shippingOption.calculated_price
 
+          const locationId =
+            shippingOption.service_zone.fulfillment_set.location.id
+
+          const itemsAtLocationWithoutAvailableQuantity = cart.items.filter(
+            (item) => {
+              if (!item.variant.manage_inventory) {
+                return false
+              }
+
+              return item.variant.inventory_items.some((inventoryItem) => {
+                if (!inventoryItem.inventory.requires_shipping) {
+                  return false
+                }
+
+                const level = inventoryItem.inventory.location_levels.find(
+                  (locationLevel) => {
+                    return locationLevel.location_id === locationId
+                  }
+                )
+
+                return !level ? true : level.available_quantity === 0
+              })
+            }
+          )
+
           return {
             ...shippingOption,
             amount: price?.calculated_amount,
             is_tax_inclusive: !!price?.is_calculated_price_tax_inclusive,
+            has_missing_inventory:
```

Comment:
nit: Can we change this to `insufficient_inventory: true`? It's nitpicky, but you cannot have missing inventory 😄

Reply:
😅 done.